File: qucore/js/src/main/scala/com/publicscript/qucore/EntityPickupNailgun.scala
package com.publicscript.qucore

import com.publicscript.qucore.MathUtils.Vec3
import com.publicscript.qucore.Resources.{model_pickup_nailgun,sfx_pickup}

class EntityPickupNailgun(world:World, apos:Vec3) extends EntityPickup(world, apos) {

  this.texture = Some(12)
  this.model = Some(model_pickup_nailgun)

  override def update() = {
    this.yaw += 0.02
    super.update()
  }

  def pickup() = {
    world.audio_play(sfx_pickup)
    world.player.weapons.addOne(new ItemWeaponNailgun(world))
    world.player.weapon_index = world.player.weapons.length - 1
    this.kill()
  }

}

File: q1k3-master/q1k3-master/source/entity_pickup_nailgun.js

class entity_pickup_nailgun_t extends entity_pickup_t {
	_init() {
		super._init();
		this._texture = 12;
		this._model = model_pickup_nailgun;
	}

	_update() {
		this._yaw += 0.02;
		super._update();
	}

	_pickup() {
		audio_play(sfx_pickup);
		game_entity_player._weapon_index = game_entity_player._weapons.push(new weapon_nailgun_t) - 1;
		this._kill();
	}
}
